services/wallet/router/routes/router_path.go

```diff
@@ -12,59 +13,73 @@ import (
 
-	TxBaseFee     *hexutil.Big // Base fee for the transaction (in ETH WEI)
-	TxPriorityFee *hexutil.Big // Priority fee for the transaction (in ETH WEI)
-	TxGasAmount   uint64       // Gas used for the transaction
-	TxBonderFees  *hexutil.Big // Bonder fees for the transaction - used for Hop bridge (in selected token)
-	TxTokenFees   *hexutil.Big // Token fees for the transaction - used for bridges (represent the difference between the amount in and the amount out, in selected token)
+	TxNonce         *hexutil.Uint64 // Nonce for the transaction
+	TxMaxFeesPerGas *hexutil.Big    // Max fees per gas (determined by client via GasFeeMode, in ETH WEI)
+	TxBaseFee       *hexutil.Big    // Base fee for the transaction (in ETH WEI)
+	TxPriorityFee   *hexutil.Big    // Priority fee for the transaction (in ETH WEI)
+	TxGasAmount     uint64          // Gas used for the transaction
+	TxBonderFees    *hexutil.Big    // Bonder fees for the transaction - used for Hop bridge (in selected token)
+	TxTokenFees     *hexutil.Big    // Token fees for the transaction - used for bridges (represent the difference between the amount in and the amount out, in selected token)
+	TxEstimatedTime fees.TransactionEstimation
 
 	TxFee   *hexutil.Big // fee for the transaction (includes tx fee only, doesn't include approval fees, l1 fees, l1 approval fees, token fees or bonders fees, in ETH WEI)
 	TxL1Fee *hexutil.Big // L1 fee for the transaction - used for for transactions placed on L2 chains (in ETH WEI)
 
 	ApprovalRequired        bool            // Is approval required for the transaction
 	ApprovalAmountRequired  *hexutil.Big    // Amount required for the approval transaction
 	ApprovalContractAddress *common.Address // Address of the contract that needs to be approved
+	ApprovalTxNonce         *hexutil.Uint64 // Nonce for the transaction
+	ApprovalMaxFeesPerGas   *hexutil.Big    // Max fees per gas (determined by client via GasFeeMode, in ETH WEI)
 	ApprovalBaseFee         *hexutil.Big    // Base fee for the approval transaction (in ETH WEI)
 	ApprovalPriorityFee     *hexutil.Big    // Priority fee for the approval transaction (in ETH WEI)
 	ApprovalGasAmount       uint64          // Gas used for the approval transaction
+	ApprovalEstimatedTime   fees.TransactionEstimation
 
 	ApprovalFee   *hexutil.Big // Total fee for the approval transaction (includes approval tx fees only, doesn't include approval l1 fees, in ETH WEI)
 	ApprovalL1Fee *hexutil.Big // L1 fee for the approval transaction - used for for transactions placed on L2 chains (in ETH WEI)
 
 	TxTotalFee *hexutil.Big // Total fee for the transaction (includes tx fees, approval fees, l1 fees, l1 approval fees, in ETH WEI)
 
-	EstimatedTime fees.TransactionEstimation
-
 	RequiredTokenBalance  *big.Int // (in selected token)
 	RequiredNativeBalance *big.Int // (in ETH WEI)
 	SubtractFees          bool
 }
 
+func (p *Path) PathIdentity() string {
```

Comment:
ugh we've got this as well a couple lines below
![image](https://github.com/user-attachments/assets/59ce72b9-7ea9-498b-b873-cff17d2bde14)

it's only used to find transaction details for a given path, so I feel we can just use the new one everywhere and remove the other function to avoid confusion
<img width="837" alt="image" src="https://github.com/user-attachments/assets/70c991e8-f009-499f-99da-00a0a7b3b90c" />

Reply:
Agree, updated.